File: 백준/Gold/23740. 버스 노선 개편하기/버스 노선 개편하기.cc
#include <bits/stdc++.h>

#define endl '\n'
#define FASTIO ios::sync_with_stdio(0), cin.tie(0), cout.tie(0)

void _init() {
#ifndef BOJ
    freopen("boj.in", "r", stdin);
#endif // BOJ
}

using namespace std;

using vi = vector<int>;

int main() {
    FASTIO;
    _init();

    int n; cin >> n;

    vector<vi> arr(n, vi(3));
    for (int i = 0; i < n; i++) {
        cin >> arr[i][0] >> arr[i][1] >> arr[i][2];
    }

    sort(arr.begin(), arr.end());
    
    vector<vi> ans;
    int left = arr[0][0], right = arr[0][1], cost = arr[0][2];
    for (int i = 1; i < n; i++) {
        int s = arr[i][0], e = arr[i][1], c = arr[i][2];
        if (s > right) {
            vi cur(3);
            cur[0] = left, cur[1] = right, cur[2] = cost;
            ans.push_back(cur);
            left = s, right = e, cost = c;
        }
        else {
            right = max<int>(right, e);
            cost = min<int>(cost, c);
        }
    }
    vi cur(3);
    cur[0] = left, cur[1] = right, cur[2] = cost;
    ans.push_back(cur);

    cout << ans.size() << endl;
    for (auto i : ans) {
        cout << i[0] << ' ' << i[1] << ' ' << i[2] << endl;
    }

    return 0;
}

File: 백준/Gold/23740. 버스 노선 개편하기/버스 노선 개편하기.py
import sys
from random import random

input = lambda: sys.stdin.readline().rstrip()

try:
    sys.stdin = open('boj.in', 'r', encoding='utf-8')
    # sys.stdout = open('boj.out', 'w', encoding='utf-8')
    pass
except FileNotFoundError:
    pass


n = int(input())
arr = [list(map(int, input().split())) for _ in range(n)]

arr.sort()

# print(arr)

ans = []

left, right, cost = arr[0]
for s, e, c in arr[1:]:
    if s > right:
        ans.append([left, right, cost])
        left = s
        right = e
        cost = c
    else:
        right = max(right, e)
        cost = min(cost, c)
ans.append([left, right, cost])

print(len(ans))
for i in ans:
    print(*i)
